ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalyzer.java

```diff
@@ -359,5 +359,46 @@ public Void visitQualifiedColumnReference(
     ) {
       throw new UnsupportedOperationException("Should of been converted to unqualified");
     }
+
+    @Override
+    public Void visitSubscriptExpression(final SubscriptExpression node, final Void context) {
+      if (groupBy.contains(node)) {
+        foundExpressionInGroupBy = true;
+      }
+      super.visitSubscriptExpression(node, context);
+      foundExpressionInGroupBy = false;
+      return null;
+    }
+
+    @Override
+    public Void visitArithmeticBinary(final ArithmeticBinaryExpression node, final Void context) {
```

Comment:
**tl;dr** this comment is a little hard to follow because I was understanding the problem as I wrote the comment. basically we need to make sure that only the _first_ node that sets `foundExpressionInGroupBy=true` can set it back to `false`.

---

so here's a freak scenario:
```sql
SELECT as_value(a['idx'] + b['idx']) FROM s GROUP BY a['idx'], a['idx'] + b['idx']
```

visit `a['idx'] + b['idx']` and set `foundExpressionInGroupBy = true`
visit (left)  `a['idx']` and set `foundExpressionInGroupBy = true`
visit `'idx'`
go back to `a['idx']` and set `foundExpressionInGroupBy = false`
visit (right) `b['idx']` and fail because `b['idx']` is not in group by

I think there are actually a few bugs here:
1. is what I showed with the weird group by scenario
1. there is another bug where we shouldn't set `foundExpressionInGroupBy` to `false` unless _we_ set it to true

that (number 2 above) could be an even simpler bug:

```sql
-- a is map<string, map<string, string>>
SELECT as_value(a[b][c]) FROM s GROUP BY a[b][c];
```

we set `foundExpressionInGroupBy` to `true` when we visit `a[b][c]` but then set it to false when we visit `a[b]` causing us to fail when we visit `[c]`

I think the solution here is to have a more complicated `foundExpressionInGroupBy` that once it is set, nobody else can set/unset it until we get back to whoever it was that first set it. You can use a tuple of `(foundExpressionInGroupBy, theGroupByExpression)` to achieve this

please add tests for this too!